src/Evolution/Systems/RelativisticEuler/Valencia/System.hpp

```diff
@@ -21,14 +31,45 @@ namespace RelativisticEuler {
 /// Zanotti or http://iopscience.iop.org/article/10.1086/303604
 namespace Valencia {
 
-template <size_t Dim>
+template <size_t Dim, typename EquationOfStateType>
 struct System {
   static constexpr bool is_in_flux_conservative_form = true;
   static constexpr bool has_primitive_and_conservative_vars = true;
   static constexpr size_t volume_dim = Dim;
+  static constexpr size_t thermodynamic_dim =
+      EquationOfStateType::thermodynamic_dim;
 
-  using variables_tag =
-      Tags::Variables<tmpl::list<TildeD, TildeTau, TildeS<Dim>>>;
+  using primitive_variables_tag = ::Tags::Variables<
+      tmpl::list<hydro::Tags::RestMassDensity<DataVector>,
+                 hydro::Tags::SpecificInternalEnergy<DataVector>,
+                 hydro::Tags::LorentzFactor<DataVector>,
+                 hydro::Tags::SpecificEnthalpy<DataVector>,
+                 hydro::Tags::Pressure<DataVector>,
+                 hydro::Tags::SpatialVelocity<DataVector, Dim>>>;
+  using variables_tag = ::Tags::Variables<
+      tmpl::list<Tags::TildeD, Tags::TildeTau, Tags::TildeS<Dim>>>;
+
+  using spacetime_variables_tag =
+      ::Tags::Variables<gr::tags_for_hydro<Dim, DataVector>>;
+
+  using conservative_from_primitive = ConservativeFromPrimitive<Dim>;
+  using primitive_from_conservative =
```

Comment:
Are there not multiple recovery schemes that are used like in the GRMHD case? If so, why are we restricting ourselves to only one when GRMHD supports multiple?

Reply:
The recovery scheme in develop is widely the most popular one, but certainly not the only one. I'd be happy to explore and code up more options, as well as updating the interface _à la_ `GrMhd` in a future PR.

Reply:
Okay, I wasn't sure since for GRMHD everybody uses a series of schemes. Thanks!